service/history/transferQueueProcessorBase.go

```diff
@@ -271,34 +280,38 @@ func getVisibilityMessageForCloseExecution(domainID string, execution workflow.W
 		es.CloseStatus:   {Type: &es.FieldTypeInt, IntData: common.Int64Ptr(int64(closeStatus))},
 		es.HistoryLength: {Type: &es.FieldTypeInt, IntData: common.Int64Ptr(historyLength)},
 	}
+	if len(memo) != 0 {
+		memoBytes, _ := json.Marshal(memo)
+		fields[es.Memo] = &indexer.Field{
+			Type: &es.FieldTypeBinary, BinaryData: memoBytes,
+		}
+	}
 
 	msg := &indexer.Message{
 		MessageType: &msgType,
 		DomainID:    common.StringPtr(domainID),
 		WorkflowID:  common.StringPtr(execution.GetWorkflowId()),
 		RunID:       common.StringPtr(execution.GetRunId()),
 		Version:     common.Int64Ptr(taskID),
-		IndexAttributes: &indexer.IndexAttributes{
-			Fields: fields,
-		},
+		Fields:      fields,
 	}
 	return msg
 }
 
-func getWorkflowExecutionTimestamp(msBuilder mutableState) time.Time {
+func getWorkflowExecutionTimestampAndMemo(msBuilder mutableState) (time.Time, map[string][]byte, bool) {
```

Comment:
May be return a struct from this method? What is the bool output meaning?

Reply:
Updated. Used separate functions for clarification. 
Return bool is to indicate whether getStartEvent success or not.